pulser/waveforms.py

```diff
@@ -517,6 +616,15 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"BlackmanWaveform({self._duration} ns, Area: {self._area:.3g})"
 
+    def __getitem__(self, index_or_slice:
+                    Union[int, slice]) -> Union[float, np.ndarray]:
+        if isinstance(index_or_slice, slice):
+            self._check_slice(index_or_slice)
+            return cast(np.ndarray, self.samples[index_or_slice])
+        else:
+            self._check_index(index_or_slice)
+            return cast(float, self.samples[index_or_slice])
```

Comment:
This is minutia, but if the `index` is `0` or `duration-1` you can return `0` instead.

Reply:
> This is minutia, but if the `index` is `0` or `duration-1` you can return `0` instead.

👍